### DISCUSSION TITLE
A few questions about file navigation

### DISCUSSION THREAD

**Discussion by josephcrawfordSRH** [ibhagwan/fzf-lua#1602] [Q&A]
I have been using Telescope for a while now in LazyVim and had it setup so that I could search for a file and then ctrl-s and type a function/property name, this would search the symbols in the selected file and I could hit enter to jump directly to the selected function/property.

In VSCode I used to just type Filename@meth and jump directly there.

I am also looking at how I could jump to a certain line in a file, for instance typing FileName:100 would open the file at line 200.

I had all of this working in Telescope but since LazyVim changed the deault to fzf-lua I figured I would give it a shot.

are there comparable capabilities with fzf-lua?

**Comment by ibhagwan** [comment]
> In VSCode I used to just type Filename@meth and jump directly there.
> I am also looking at how I could jump to a certain line in a file, for instance typing FileName:100 would open the file at line 200.

Unfortunately this is not easily achievable with fzf-lua due it being run in an external process (fzf), that makes it so once all the data is handed over to fzf the interaction between the process and fzf-lua remains minimal (otherwise you’d have perf issues hanging on the neovim main thread).

There could be semi-solution for this with query triggers that attempt to mimic this functionality but this will never be as easy as in Telescope due to the above.

**Comment by josephcrawfordSRH** [comment]
@ibhagwan thank you for that information.  I did find a way to open document symbols so i can jump around the current document, may just take two steps find file, jump to symbol. It's a bit of a pain but may be the only way.

**Reply by ibhagwan** [reply]
> @ibhagwan thank you for that information. I did find a way to open document symbols so i can jump around the current document, may just take two steps find file, jump to symbol. It's a bit of a pain but may be the only way.

There are may ways to navigate quickly, also try `lsp_finder`, `treesitter` and `blines`, I would probably make a toggle to switch between these.
